### PR TITLE
[archive] use threads for xz compression

### PR DESCRIPTION
**Pull request by pmoravec**
Resolves: #1196

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/archive.py [hunk 1]
```diff
@@ -133,9 +133,10 @@ class FileCacheArchive(Archive):
     _archive_root = ""
     _archive_name = ""
 
-    def __init__(self, name, tmpdir):
+    def __init__(self, name, tmpdir, threads):
```
Comment (bmr-cymru) [inline sos/archive.py]: You can't just stuff this into `FileCacheArchive`; it's part of a class hierarchy that defines interfaces: they all need the same initialiser arguments.

**Comment by pmoravec** [comment]
Finally fixed travis failures, now the PR should be OK.

File: sos/sosreport.py [hunk 2]
```diff
@@ -757,11 +757,22 @@ def get_temp_file(self):
     def _set_archive(self):
         archive_name = os.path.join(self.tmpdir,
                                     self.policy.get_archive_name())
+        # determine number of threads to use for compressing - applicable
+        # only for xz and of version 5.2 or higher
+        try:
+            xz_version = self.policy.package_manager\
+                         .all_pkgs()['xz']["version"]
```
Comment (BryanQuigley) [inline sos/sosreport.py]: xz-utils is the package name on Ubuntu/Debian.

**Comment by pmoravec** [comment]
Updated PR according to the archive/policy discussion.

**Pull request closed without merging**